File: src/Shared/Logistics.Shared.Models/PaymentDto.cs
using Logistics.Shared.Enums;

namespace Logistics.Shared.Models;

public class PaymentDto
{
    public string Id { get; set; } = default!;
    public DateTime CreatedDate { get; set; }
    public DateTime? PaymentDate { get; set; }
    public PaymentMethod? Method { get; set; }
    public decimal Amount { get; set; }
    public PaymentStatus Status { get; set; }
    public PaymentFor PaymentFor { get; set; }
    public AddressDto? BillingAddress { get; set; }
    public string? Comment { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/models/payment/PaymentDto.ts
import {PaymentFor, PaymentMethod, PaymentStatus} from '@core/enums';
import {AddressDto} from '../AddressDto';


export interface PaymentDto {
  id: string;
  createdDate: string;
  paymentDate?: string;
  method?: PaymentMethod;
  amount: number;
  status: PaymentStatus;
  paymentFor: PaymentFor;
  billingAddress?: AddressDto;
  comment?: string;
}
